classes/woocommerce/class-account-details-avatar.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Automattic\Gravatar\GravatarEnhanced\Woocommerce;
+
+class AccountDetailsAvatar {
+	private $current_user;
+	private $user_email;
+	private $display_name;
+	private $current_user_locale;
+
+	public function init() {
+		add_action( 'woocommerce_before_account_navigation', [ $this, 'start_capture_page' ] );
+		add_action( 'woocommerce_after_account_navigation', [ $this, 'end_capture_page' ] );
+	}
+
+	private function get_user_data() {
```

Comment:
This is a `get` function which doesn't return any data but instead puts the values on member variables, and also performs escaping. I think it might be clearer if the function returned the values, and the escaping can then happen as near to the output as possible.

It may be simpler to just call `wp_get_current_user();` directly when needed as the function doesn't really provide anything extra, with the locale stuff being split into it's own function.